File: 필수문항/111_자리배정.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int C = Integer.parseInt(st.nextToken());
        int R = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(br.readLine());
        if (K > C * R) {
            System.out.println(0);
        } else {
            int[][] graph = new int[C * R + 1][2];
            graph[0][0] = 1;
            int[] wall = { R, C, 1, 1 };
            int pointer = 0;

            for (int num = 1; num <= R * C; num++) {
                switch (pointer) {
                    case 0:
                        graph[num][0] = graph[num - 1][0];
                        graph[num][1] = graph[num - 1][1] + 1;
                        if (graph[num][1] == wall[pointer]) {
                            pointer = 1;
                            wall[3] += 1;
                        }
                        break;
                    case 1:
                        graph[num][1] = graph[num - 1][1];
                        graph[num][0] = graph[num - 1][0] + 1;
                        if (graph[num][0] == wall[pointer]) {
                            pointer = 2;
                            wall[0] -= 1;
                        }
                        break;
                    case 2:
                        graph[num][0] = graph[num - 1][0];
                        graph[num][1] = graph[num - 1][1] - 1;
                        if (graph[num][1] == wall[pointer]) {
                            pointer = 3;
                            wall[1] -= 1;
                        }
                        break;
                    case 3:
                        graph[num][1] = graph[num - 1][1];
                        graph[num][0] = graph[num - 1][0] - 1;
                        if (graph[num][0] == wall[pointer]) {
                            pointer = 0;
                            wall[2] += 1;
                        }
                        break;
                }
            }
            System.out.printf("%d %d", graph[K][0], graph[K][1]);
        }

    }
}

File: 필수문항/111_자리배정.py
import sys

input = sys.stdin.readline

C, R = map(int, input().split())
K = int(input())

if K > C * R:
    print(0)
else:
    graph = [[0, 0] for _ in range(C * R + 1)]
    graph[0][0] = 1
    wall = [R, C, 1, 1]  # top, right, down, left
    pointer = 0

    for num in range(1, C * R + 1):
        if pointer == 0:
            x, y = graph[num - 1]
            graph[num][0] = x
            graph[num][1] = y + 1
            if graph[num][1] == wall[pointer]:
                pointer = 1
                wall[3] += 1
        elif pointer == 1:
            x, y = graph[num - 1]
            graph[num][1] = y
            graph[num][0] = x + 1
            if graph[num][0] == wall[pointer]:
                pointer = 2
                wall[0] -= 1
        elif pointer == 2:
            x, y = graph[num - 1]
            graph[num][0] = x
            graph[num][1] = y - 1
            if graph[num][1] == wall[pointer]:
                pointer = 3
                wall[1] -= 1
        else:
            x, y = graph[num - 1]
            graph[num][1] = y
            graph[num][0] = x - 1
            if graph[num][0] == wall[pointer]:
                pointer = 0
                wall[2] += 1

    print(*graph[K], sep=" ")
